app/src/main/java/io/capibaras/abcall/viewmodels/LoginViewModel.kt

```
@@ -4,48 +4,109 @@ import androidx.compose.runtime.getValue
+                        errorUIState = ErrorUIState.Error(R.string.error_authenticate)
+                    }
+                } else {
+                    errorUIState = if (response.code() == 401) {
+                        ErrorUIState.Error(R.string.error_incorrect_credentials)
+                    } else {
+                        val errorBody = response.errorBody()?.string()
+                        if (errorBody != null) {
+                            try {
+                                val jsonObject = JSONObject(errorBody)
+                                jsonObject.getString("message")
+                            } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)